astropy/table/column.py

```diff
@@ -131,21 +78,19 @@ def col_iter_str_vals(col):
 def col_copy(col):
     """
     This is a mixin-safe version of Column.copy() (with copy_data=True).
-
-    Warning: this function is subject to change or removal.
     """
     if isinstance(col, BaseColumn):
         return col.copy()
 
-    if hasattr(col, '_astropy_column_attrs'):
-        col_setattr(col, 'parent_table', None)  # Don't copy weakref to parent table
-    newcol = col.copy() if hasattr(col, 'copy') else deepcopy(col)
+    parent_table = col.info.parent_table
+    col.info.parent_table = None
```

Comment:
Since it took a lot of time to figure out it was necessary, maybe add a comment here why you are temporarily removing the weakref and doing the `try/finally`?